### PR TITLE
MULE-9129 - Cleanup transport infrastructure

### PR DESCRIPTION
**Pull request by pablolagreca**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/ExceptionListenerTestCase.java [hunk 1]
```diff
@@ -29,24 +27,24 @@
...
-        list.add(Mockito.mock(OutboundEndpoint.class));
+        list.add(Mockito.mock(MessageProcessor.class));
+        list.add(Mockito.mock(MessageProcessor.class));
 
         AbstractExceptionListener router = new DefaultMessagingExceptionStrategy(null);
         assertNotNull(router.getMessageProcessors());
         assertEquals(0, router.getMessageProcessors().size());
 
-        router.addEndpoint(Mockito.mock(OutboundEndpoint.class));
+        router.addEndpoint(Mockito.mock(MessageProcessor.class));
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/ExceptionListenerTestCase.java]: shouldn't the name of this method also change?

File: core-tests/src/test/java/org/mule/MuleEventTestCase.java [hunk 2]
```diff
@@ -147,17 +138,17 @@ public void testEventSerializationRestart() throws Exception
 
     }
 
+    @Ignore("Credentials were provided by the inbound endpoint")
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/MuleEventTestCase.java]: seems to be than rather ignoring, this test should go away. All credentials will now be handled at the ConnectionProvider level right?

File: core-tests/src/test/java/org/mule/api/transport/ExceptionHandlingReplyToHandlerDecoratorTestCase.java [hunk 3]
```diff
@@ -14,6 +14,7 @@
 import org.mule.api.MessagingException;
 import org.mule.api.MuleEvent;
 import org.mule.api.exception.MessagingExceptionHandler;
+import org.mule.execution.ExceptionHandlingReplyToHandlerDecorator;
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/api/transport/ExceptionHandlingReplyToHandlerDecoratorTestCase.java]: this seems unnecesary

File: core-tests/src/test/java/org/mule/processor/AsyncInterceptingMessageProcessorTestCase.java [hunk 4]
```diff
@@ -64,10 +64,11 @@ protected void doSetUp() throws Exception
         messageProcessor = createAsyncInterceptingMessageProcessor(target);
     }
 
+    @Ignore
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/processor/AsyncInterceptingMessageProcessorTestCase.java]: why?

File: core-tests/src/test/java/org/mule/processor/LaxAsyncInterceptingMessageProcessorTestCase.java [hunk 5]
```diff
@@ -6,21 +6,23 @@
...
 import org.mule.tck.testmodels.mule.TestTransaction;
 import org.mule.transaction.TransactionCoordination;
 
+import org.junit.Ignore;
 import org.junit.Test;
 
 public class LaxAsyncInterceptingMessageProcessorTestCase extends AsyncInterceptingMessageProcessorTestCase
 {
+
+    @Ignore
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/processor/LaxAsyncInterceptingMessageProcessorTestCase.java]: same as before

File: core-tests/src/test/java/org/mule/processor/LaxSedaStageInterceptingMessageProcessorTestCase.java [hunk 6]
```diff
@@ -6,23 +6,24 @@
...
 import org.mule.transaction.TransactionCoordination;
 
+import org.junit.Ignore;
 import org.junit.Test;
 
 public class LaxSedaStageInterceptingMessageProcessorTestCase extends
     SedaStageInterceptingMessageProcessorTestCase
 {
 
+    @Ignore
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/processor/LaxSedaStageInterceptingMessageProcessorTestCase.java]: same

File: core-tests/src/test/java/org/mule/registry/RegistryBrokerTestCase.java [hunk 7]
```diff
@@ -62,37 +59,13 @@ public void testCrossRegistryLifecycleOrder() throws MuleException
...
+        assertEquals("flow2-start flow-start ", tracker.toString());
 
         tracker = new String();
         muleContext.stop();
 
         // Both services are stopped before either connector
-        assertEquals("flow2-stop flow-stop conn2-stop conn-stop ", tracker);
-    }
-
-    class LifecycleTrackerConnector extends TestConnector
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/registry/RegistryBrokerTestCase.java]: why should this change remove this?

File: core-tests/src/test/java/org/mule/MuleEventTestCase.java (hunk 2 above)
Comment (pablolagreca) [inline core-tests/src/test/java/org/mule/MuleEventTestCase.java]: Removing and adding to misc mule 4 issues to rethink test case

File: core-tests/src/test/java/org/mule/ExceptionListenerTestCase.java (hunk 1 above)
Comment (pablolagreca) [inline core-tests/src/test/java/org/mule/ExceptionListenerTestCase.java]: Changed to addMessageProcessor

File: core-tests/src/test/java/org/mule/api/transport/ExceptionHandlingReplyToHandlerDecoratorTestCase.java (hunk 3 above)
Comment (pablolagreca) [inline core-tests/src/test/java/org/mule/api/transport/ExceptionHandlingReplyToHandlerDecoratorTestCase.java]: it was, but moved the test to a different package so I can remove the import. Also the package didn't make sense, was called transport.

**Pull request closed without merging**